src/component/context/ContextList/ContextList.tsx

```diff
@@ -1,42 +1,133 @@
-import { useContext, useState, VFC } from 'react';
-import { Add, Album, Delete, Edit } from '@mui/icons-material';
-import { Link, useNavigate } from 'react-router-dom';
+import { useContext, useMemo, useState, VFC } from 'react';
+import { useGlobalFilter, useSortBy, useTable } from 'react-table';
 import {
-    Button,
-    IconButton,
-    List,
-    ListItem,
-    ListItemIcon,
-    ListItemText,
-    Tooltip,
-    useMediaQuery,
-} from '@mui/material';
+    Table,
+    SortableTableHeader,
+    TableBody,
+    TableCell,
+    TableRow,
+    TablePlaceholder,
+    TableSearch,
+} from 'component/common/Table';
 import { PageContent } from 'component/common/PageContent/PageContent';
 import { PageHeader } from 'component/common/PageHeader/PageHeader';
 import { ConditionallyRender } from 'component/common/ConditionallyRender/ConditionallyRender';
-import {
-    CREATE_CONTEXT_FIELD,
-    DELETE_CONTEXT_FIELD,
-    UPDATE_CONTEXT_FIELD,
-} from 'component/providers/AccessProvider/permissions';
+import { UPDATE_CONTEXT_FIELD } from 'component/providers/AccessProvider/permissions';
 import { Dialogue as ConfirmDialogue } from 'component/common/Dialogue/Dialogue';
 import AccessContext from 'contexts/AccessContext';
 import useUnleashContext from 'hooks/api/getters/useUnleashContext/useUnleashContext';
 import useContextsApi from 'hooks/api/actions/useContextsApi/useContextsApi';
 import useToast from 'hooks/useToast';
 import { formatUnknownError } from 'utils/formatUnknownError';
-import { useStyles } from './styles';
+import { AddContextButton } from './AddContextButton/AddContextButton';
+import { SearchHighlightProvider } from 'component/common/Table/SearchHighlightContext/SearchHighlightContext';
+import { sortTypes } from 'utils/sortTypes';
+import { LinkCell } from 'component/common/Table/cells/LinkCell/LinkCell';
+import { ContextActionsCell } from './ContextActionsCell/ContextActionsCell';
+import { Adjust } from '@mui/icons-material';
+import { Box } from '@mui/material';
 
 const ContextList: VFC = () => {
     const { hasAccess } = useContext(AccessContext);
     const [showDelDialogue, setShowDelDialogue] = useState(false);
-    const smallScreen = useMediaQuery('(max-width:700px)');
     const [name, setName] = useState<string>();
-    const { context, refetchUnleashContext } = useUnleashContext();
+    const { context, refetchUnleashContext, loading } = useUnleashContext();
     const { removeContext } = useContextsApi();
     const { setToastData, setToastApiError } = useToast();
-    const navigate = useNavigate();
-    const { classes: styles } = useStyles();
+    const data = useMemo(() => {
```

Comment:
I think we should talk about these kinds of optimisations. In this case, I think actually using `useMemo` is hurting us more than it's helping us. I don't see how this component will re-render, except perhaps if you trigger useSWRs focus refetch by changing tabs to refetch the context fields from the API. 

But even if this was a component that changed often, how many context fields would users create? 10? 100? Even at those numbers, I think useMemo would still not be worth it. By adding useMemo we are adding complexity, memory usage (memoized data will not be garbage collected), and not gaining a whole lot in the process.  

It _sometimes_ makes sense to use `useMemo`, if the computation is heavy - but there's always a tradeoff and I want us to be conscious about when we use it, and what kind of benefit we get in return. 

This article is a good reference for a discussion: https://kentcdodds.com/blog/usememo-and-usecallback